app/src/main/java/chat/rocket/android/chatroom/presentation/ChatRoomPresenter.kt

```diff
@@ -553,4 +554,33 @@ class ChatRoomPresenter @Inject constructor(private val view: ChatRoomView,
             }
         }
     }
-}
+
+    /**
+     * Save unfinished message, when user left chat room without sending a message. It also clears
+     * saved message from local repository when unfinishedMessage is blank.
+     *
+     * @param chatRoomId Chat room Id.
+     * @param unfinishedMessage The unfinished message to save.
+     */
+    fun saveUnfinishedMessage(chatRoomId: String, unfinishedMessage: String) {
+        val key = LocalRepository.UNFINISHED_MSG_KEY + chatRoomId
+        if (unfinishedMessage.isNotBlank()) {
+            localRepository.save(key, unfinishedMessage)
+        } else {
+            localRepository.clear(key)
+        }
+    }
+
+    /**
+     * Get unfinished message from local repository, when user left chat room without
+     * sending a message and now the user is back.
+     *
+     * @param chatRoomId Chat room Id.
+     *
+     * @return Returns the unfinished message.
+     */
+    fun getUnfinishedMessage(chatRoomId: String): String {
+        val key = LocalRepository.UNFINISHED_MSG_KEY + chatRoomId
```

Comment:
The same here:
`val key = "${currentServer}_${LocalRepository.UNFINISHED_MSG_KEY}$chatRoomId"`